### ISSUE TITLE
Compliance UI: Syntax Error: Invalid number, expected digit but got: "c".

### ISSUE THREAD

**Issue by vmwiz** [stackrox/stackrox#4342]
Since upgrade to StackRox 3.73.1, the Compliance tab in the UI gives the following error message.

Error message:
```
Syntax Error: Invalid number, expected digit but got: "c".
```

Stack trace
```
GraphQLError: Syntax Error: Invalid number, expected digit but got: "c".
    at w (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:762298)
    at R (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:768089)
    at I (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:767574)
    at T (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:766400)
    at t.lookahead (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:764998)
    at t.advance (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:764877)
    at t.expectToken (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:785673)
    at t.parseName (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:770077)
    at t.parseOperationDefinition (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:771182)
    at t.parseDefinition (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:770436)
    at t.many (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:786671)
    at t.parseDocument (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:770239)
    at n.experimentalFragmentVariables (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:787927)
    at $ (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:787943)
    at Y (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:788562)
    at p (https://central-stackrox.apps.ocp.example.com/static/js/5218.7852c2ae.chunk.js:1:7941)
    at ve (https://central-stackrox.apps.ocp.example.com/static/js/2078.b97dd7a4.chunk.js:1:8912)
    at sa (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:973247)
    at $s (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1025547)
    at Tu (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1012788)
    at Pu (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1012716)
    at Cu (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1012579)
    at bu (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1009545)
    at https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:959039
    at t.unstable_runWithPriority (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1823387)
    at Go (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:958816)
    at Ko (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:958984)
    at Yo (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:958919)
    at hu (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1006922)
    at Aa (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:978572)
    at Object.t [as next] (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1981960)
    at O (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2061641)
    at w (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2062147)
    at t.next (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2062680)
    at https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1923236
    at Array.forEach (<anonymous>)
    at w (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1923204)
    at t.reportResult (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1932978)
    at Object.next (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1932557)
    at https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1923236
    at Array.forEach (<anonymous>)
    at w (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1923204)
    at Object.next (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1923727)
    at O (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2061641)
    at w (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2062147)
    at t.next (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2062680)
    at https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1922739
```

Component stack
```
    at ve (https://central-stackrox.apps.ocp.example.com/static/js/2078.b97dd7a4.chunk.js:1:8569)
    at Ce (https://central-stackrox.apps.ocp.example.com/static/js/2078.b97dd7a4.chunk.js:1:11955)
    at div
    at div
    at section
    at _e (https://central-stackrox.apps.ocp.example.com/static/js/2078.b97dd7a4.chunk.js:1:15440)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1460237)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1461154)
    at tr
    at r (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2089904)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1460237)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1461154)
    at n (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:23443)
    at n (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1461691)
    at div
    at it (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2097754)
    at main
    at div
    at n (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:347088)
    at div
    at div
    at Ce (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2089230)
    at ir (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2124516)
    at sr (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2125762)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1460237)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1461154)
    at wi
    at n (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:23443)
    at n (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1461691)
    at p (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:25497)
    at t (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1457574)
    at l (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:559455)
    at o (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:560900)
    at m (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1425166)
    at s (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:2076430)
    at l (https://central-stackrox.apps.ocp.example.com/static/js/main.0028f9d6.js:2:1423215)
 ```
 
 We tried a fresh install of 3.73.1, deleting the database, redeploying new sensors on the clients with the same effect.

**Comment by johannes94** [comment]
Hello @vmwiz, thanks for reporting this issue. This looks like the UI gets a response it can not parse. Unfortunately this error message doesn't provide enough context to diagnose the problem.

If it is possible for you, you can use the Debug Window of your browser to see the actual response returned by the API when opening the compliance tab in the UI and share it with us.

**Comment by vmwiz** [comment]
Hi @johannes94 ; here's a screenshot of the error console showing the same error as in the UI

![image](https://user-images.githubusercontent.com/17565465/212640196-0da93caf-940b-4ce7-a2a8-434265b48707.png)

Which API call would you be interested into? All graphql calls seems to return valid data's (well formed json with proper data's)

**Comment by vjwilson** [comment]
@vmwiz That error screen is not showing the API response. Those will be in the browser dev tools Network tab.

As soon as you try to load the Compliance page, you will see several Fetch/XHR calls that `graphql?opname=...`. Please look at each of those responses to see if any of those have error messages, or some other string other than well-formed JSON data.

Here is a screenshot of what you are looking for in the Network panel. (The response I have highlighted is an example of wll-formed JSON.)
![Screen Shot 2023-01-16 at 10 34 37 AM](https://user-images.githubusercontent.com/715729/212715428-a9d044d9-ea50-4e22-98c1-4142666d7150.png)

**Comment by vmwiz** [comment]
Hi @vjwilson

I reviewed all graphql calls and they all looked legit to me (well formed JSON with relevant data's) ; in case I was wrong here are the various answers:

`graphql?opname=healths`
```
{
    "data": {
        "results": {
            "total": 8,
            "uninitialized": 0,
            "healthy": 8,
            "degraded": 0,
            "unhealthy": 0,
            "__typename": "ClusterHealthCounter"
        }
    }
}
```
`graphql?opname=summary_counts`
```
{
    "data": {
        "clusterCount": 8
    }
}
```
`graphql?opname=clustersCount`
```
{
    "data": {
        "results": 8
    }
}
```
`graphql?opname=namespacesCount`
```
{
    "data": {
        "results": 558
    }
}
```
`graphql?opname=nodesCount`
```
{
    "data": {
        "results": 105
    }
}
```
`graphql?opname=deploymentsCount`
```
{
    "data": {
        "results": 1488
    }
}
```
`graphql?opname=runStatuses`
```
{
    "data": {
        "complianceRunStatuses": {
            "invalidRunIds": [],
            "runs": [],
            "__typename": "GetComplianceRunStatusesResponse"
        }
    }
}
```
`graphql?opname=getAggregatedResultsAcrossEntity_CLUSTER`
[getAggregatedResultsAcrossEntity_CLUSTER.txt](https://github.com/stackrox/stackrox/files/10432114/getAggregatedResultsAcrossEntity_CLUSTER.txt)

`graphql?opname=getAggregatedResultsByEntity_CLUSTER`
[getAggregatedResultsByEntity_CLUSTER.txt](https://github.com/stackrox/stackrox/files/10432122/getAggregatedResultsByEntity_CLUSTER.txt)

`graphql?opname=getAggregatedResultsAcrossEntity_NAMESPACE`
[getAggregatedResultsAcrossEntity_NAMESPACE.txt](https://github.com/stackrox/stackrox/files/10432129/getAggregatedResultsAcrossEntity_NAMESPACE.txt)

`graphql?opname=getAggregatedResultsAcrossEntity_NODE`
[getAggregatedResultsAcrossEntity_NODE.txt](https://github.com/stackrox/stackrox/files/10432140/getAggregatedResultsAcrossEntity_NODE.txt)

`graphql?opname=getComplianceStandards`
[getComplianceStandards.txt](https://github.com/stackrox/stackrox/files/10432142/getComplianceStandards.txt)

I was able to trigger a compliance scan through the API and the results are showing-up in the main dashboard
![image](https://user-images.githubusercontent.com/17565465/212825574-3788eabb-a74a-41cb-a538-9f9914d35659.png)

**Comment by vjwilson** [comment]
@vmwiz There are about a dozen more GraphQL calls on that page, with opnames that start with "getAggregated...", "getCompliance...", and "complianceStandards". We need the results for those calls, as well.

**Comment by vmwiz** [comment]
I sent you all the ones that are called during the page rendering, the getAggregated ones are in my previous message (had to include them as file attachments due to the huge size of the json)

![image](https://user-images.githubusercontent.com/17565465/212932747-7b61071f-b6a5-43f1-9a37-9a9a35fb69d7.png)

**Comment by vjwilson** [comment]
@vmwiz 
I didn't find anything wrong in the rest of those files, but I noticed that your failure seems to happen right after receiving the compliance standard definitions, and you also have the Compliance Operator installed in your environment. There was an incompatibility with version 3.73.0 and the Compliance Operator, which should have been fixed with the patch release 3.73.1.

Can you verify that you indeed have 3.73.1 installed, and not still 3.73.0 ?

**Comment by vmwiz** [comment]
@vjwilson 
Damn you are right, 3.73.0 was deployed instead of 3.73.1 ; I updated the helm chart, redeployed and the compliance UI is now visible. My apologies!
